scholarx/archive/functions.js

```diff
@@ -82,21 +90,71 @@ function renderAllProfiles() {
     $("#mentorProfiles").html(mentorProfiles);
     $("#menteeProfiles").html(menteeProfiles);
 }
+function uncheckCheckboxes(){
+    for(let i=years.sort()[0]; i<=years[years.length-1]; i++){
+        document.getElementById(i).checked = false;
+    }
+    for(let i=0; i<industries.length; i++){
+        document.getElementById(industries[i].replace(/\s+/g, '_').toLowerCase()).checked = false;
+    }
+}
 function renderCohortCheckboxes(){
     const data = { checkboxes: years.map(function(year) {
         return { id: year };
     }) };
+    const industriesData = { checkboxes: industries.map(function(industry) {
+        return { id: industry , htmlId:industry.replace(/\s+/g, '_').toLowerCase()}
+    }) };
     let template = document.getElementById("cohort").innerHTML;
     let output = Mustache.render(template, data);
     document.getElementById("cohort-filters").innerHTML = output;
+    //render industry checkboxes dynamically
+    let industryTemplate = document.getElementById("industry-filter-template").innerHTML;
+    let industry = Mustache.render(industryTemplate, industriesData);
+    document.getElementById("dynamic-industry-filters").innerHTML = industry;
+}
+function filterByIndustry(industry){
+    let mentorsData = []
+    const selectedYears = years.filter((year) => document.getElementById(year).checked);
+    const selectedIndustries = industries.filter((industry) => document.getElementById(industry.replace(/\s+/g, '_').toLowerCase()).checked);
```

Comment:
```suggestion
    const selectedIndustries = industries.filter((industry) => document.getElementById(industry.replace(/\s+/g, '_').toLowerCase()).checked);
```

Move this outside the function as a global variable. @janithlahirukariyawasam